alica_engine/src/engine/AlicaTimer.cpp

```diff
@@ -29,12 +29,24 @@ class AlicaSystemTimer::AlicaSystemTimerImpl
         }
         _isActive = true;
         _thread = std::thread([this]() {
+            using namespace std::chrono;
+            auto inner_loop_duration = milliseconds(std::min(int64_t(500), _period));
             while (_isActive) {
+                auto end_time = system_clock::now().time_since_epoch() + milliseconds(_period);
+
                 _userCb();
-                int64_t sleep_duration = _period;
-                while ((sleep_duration > 0) && _isActive) {
-                    std::this_thread::sleep_for(std::chrono::milliseconds(std::min(int64_t(500), _period)));
-                    sleep_duration -= std::min(int64_t(500), _period);
+
+                auto now = system_clock::now().time_since_epoch();
+                // This second while is a trick to loop more than once, in 500ms intervals,
```

Comment:
nit: I think a neater solution would be to wait on a condition variable which can be notified immediately on Ctrl-C, can be done in another PR though

Reply:
How would you wake up after 500 seconds then?  Nobody would notify you.

Reply:
or rather, how would you wake up after the time is up

Reply:
https://en.cppreference.com/w/cpp/thread/condition_variable/wait_for

I guess it was a dumb question